File: app/src/main/java/com/example/calculator/MainActivity.java
package com.example.calculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private EditText textEdit1;
    private EditText textEdit2;
    private TextView text_equal;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        textEdit1 = findViewById(R.id.text_edit1);
        textEdit2 = findViewById(R.id.text_edit2);
        text_equal = findViewById(R.id.text_equal);

        Button buttonPlus = findViewById(R.id.button_plus);
        Button buttonMinus = findViewById(R.id.button_minus);
        Button buttonMultiply = findViewById(R.id.button_multiply);
        Button buttonDivision = findViewById(R.id.button_division);
        Button buttonClose = findViewById(R.id.button_close);


        buttonPlus.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calculate('+');
            }
        });

        buttonMinus.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calculate('-');
            }
        });

        buttonMultiply.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calculate('*');
            }
        });

        buttonDivision.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calculate('/');
            }
        });

        buttonClose.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
                System.exit(0);
            }
        });
    }

    private void calculate(char operator) {
        String number1 = textEdit1.getText().toString();
        String number2 = textEdit2.getText().toString();

        if (!number1.isEmpty() && !number2.isEmpty()) {
            double num1 = Double.parseDouble(number1);
            double num2 = Double.parseDouble(number2);

            double result = 0;

            switch (operator) {
                case '+':
                    result = num1 + num2;
                    break;
                case '-':
                    result = num1 - num2;
                    break;
                case '*':
                    result = num1 * num2;
                    break;
                case '/':
                    if (num2 != 0) {
                        result = num1 / num2;
                    } else {
                        Toast.makeText(MainActivity.this, "Cannot divide by zero", Toast.LENGTH_SHORT).show();
                    }
                    break;
            }

            text_equal.setText(String.valueOf(result));
        }
    }
}

File: app/src/main/java/com/example/calculator/MainActivity.kt
package com.example.calculator

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var textEdit1: EditText? = null
    private var textEdit2: EditText? = null
    private var text_equal: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        textEdit1 = findViewById(R.id.text_edit1)
        textEdit2 = findViewById(R.id.text_edit2)
        text_equal = findViewById(R.id.text_equal)
        val buttonPlus = findViewById<Button>(R.id.button_plus)
        val buttonMinus = findViewById<Button>(R.id.button_minus)
        val buttonMultiply = findViewById<Button>(R.id.button_multiply)
        val buttonDivision = findViewById<Button>(R.id.button_division)
        val buttonClose = findViewById<Button>(R.id.button_close)
        buttonPlus.setOnClickListener { calculate('+') }
        buttonMinus.setOnClickListener { calculate('-') }
        buttonMultiply.setOnClickListener { calculate('*') }
        buttonDivision.setOnClickListener { calculate('/') }
        buttonClose.setOnClickListener {
            finish()
            System.exit(0)
        }
    }

    private fun calculate(operator: Char) {
        val number1 = textEdit1!!.text.toString()
        val number2 = textEdit2!!.text.toString()
        if (!number1.isEmpty() && !number2.isEmpty()) {
            val num1 = number1.toDouble()
            val num2 = number2.toDouble()
            var result = 0.0
            when (operator) {
                '+' -> result = num1 + num2
                '-' -> result = num1 - num2
                '*' -> result = num1 * num2
                '/' -> if (num2 != 0.0) {
                    result = num1 / num2
                } else {
                    Toast.makeText(this@MainActivity, "Cannot divide by zero", Toast.LENGTH_SHORT)
                        .show()
                }
            }
            text_equal!!.text = result.toString()
        }
    }
}